src/search/FindInFilesUI.js

```diff
@@ -102,10 +107,45 @@ define(function (require, exports, module) {
 
                 StatusBar.hideBusyIndicator();
             })
+            .progress(function (results) {
+                // Update first page of panel (100 matches) for every file with matches,
+                // then only update the summary for every page (100 matches)
+                if (results.matches && results.matches.length) {
+                    progressCount += results.matches.length;
+                    if (!progressFirstUpdate) {
+                        // First update
+                        _resultsView.open();
+                        if (_findBar) {
+                            _findBar.close();
+                        }
+                        progressFirstUpdate = true;
+                    } else if (!progressFirstPage) {
+                        // First page
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            progressFirstPage = true;
+                        }
+                        _resultsView._updateResults();
+                    } else {
+                        // Subsequent full pages
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            _resultsView._showSummary();
```

Comment:
@redmunds I think that we should check if `SearchResultsView._currentStart + 100 > SearchModel.numMatches` and if is true we can do `_updateResults()` and if not, we can just update the summary.